### DISCUSSION TITLE
useQuery returns stale data on rendering for first time

### DISCUSSION THREAD

**Discussion by PoojaC94** [TanStack/query#4834] [Q&A]
I'm using a useQuery which is of the form:
`const { data } = useQuery(foo, async () => {...}) `

where 
1. foo is a value that changes when I perform an action, say, select a dropdown menu item, item1
2. when I do that, useQuery is called with the stale data
3. The entire functional component is re-rendered, going to the initial value
4. When I click on item1 in the dropdown again, the component is rendered with the correct value

I use staleTime: Infinity

I want the component to render the proper value immediately and not on multiple reload of component. Please let me know if using useQuery in this case would be a good idea

**Comment by TkDodo** [comment]
show a reproduction please

**Reply by PoojaC94** [reply]
It is similar to this issue https://github.com/TanStack/query/discussions/3927

**Reply by PoojaC94** [reply]
I tried the suggestion you posted. `notifyOnChangeProps: 'tracked'` but it didn't work. Is there any workaround?

**Reply by PoojaC94** [reply]
The behaviour of the component is this:

We fetch the data on page load
The user changes the dropdown menu from "foo" to "bar".

Render (First): The component renders once with the stale data.
This overwrites the value of "bar" in the state of the input with "foo".
Render (Second): The query returns successfully and "bar" now obliterates "foo".

**Reply by TkDodo** [reply]
I can't help without seeing a reproduction

**Reply by PoojaC94** [reply]
`const ids: number[] = useMemo(() => {
    return sampleArray.map((sampleOption) => Number(sampleOption.Team.Id));
  }, [sampleArray]);

  const { data: information} = useQuery(["key", ids], () => getInfo(ids), {
    staleTime: Infinity,
   enabled: !!param1
  });`

**Reply by PoojaC94** [reply]
here ids is changed whenever I click on a dropdown value(which changes sampleArray), and I need to use ids in the getInfo API

**Reply by PoojaC94** [reply]
What actually happens is that the component seems to go back to the initial state when I change the dropdown value after loading for the first time and when I click on the same dropdown value again, it renders the correct values

**Reply by TkDodo** [reply]
a runnable reproduction that actually shows the issue. codesandbox / stackblitz are good for those.

**Reply by PoojaC94** [reply]
[link](https://tkdodo.eu/blog/react-query-fa-qs#how-can-i-pass-parameters-to-refetch) I want to fetch the same query for a different id as you have explained in your blog :)

**Reply by PoojaC94** [reply]
and I want to fetch it as soon as the id/associated parameter changes
